src/course-outline/utils.jsx

```diff
@@ -0,0 +1,116 @@
+import {
+  CheckCircle as CheckCircleIcon,
+  Lock as LockIcon,
+  EditOutline as EditOutlineIcon,
+} from '@edx/paragon/icons';
+
+import { SECTION_BADGE_STATUTES, STAFF_ONLY } from './constants';
+
+/**
+ * Get section status depended on section info
+ * @param {bool} published - value from section info
+ * @param {bool} releasedToStudents - value from section info
+ * @param {bool} visibleToStaffOnly - value from section info
+ * @param {string} visibilityState - value from section info
+ * @param {bool} staffOnlyMessage - value from section info
+ * @returns {typeof SECTION_BADGE_STATUTES}
```

Comment:
```suggestion
 * @returns {SECTION_BADGE_STATUTES[keyof SECTION_BADGE_STATUTES]}
```
What you have is wrong - it's returning an `Object` when this is a `string`. Use this ^ along with the other fix I suggested and you actually get the right types:

![Screenshot 2023-12-08 at 3 12 31 PM](https://github.com/openedx/frontend-app-course-authoring/assets/945577/98976b40-a282-466a-9890-c2baaab96647)

Throw in `// @ts-check` at the top of `utils.jsx` and `SectionCard.tsx` and you can see these issues more easily in VS Code.